src/shogun2-core/src/main/java/de/terrestris/shogun2/model/wps/WpsReference.java

```diff
@@ -0,0 +1,166 @@
+package de.terrestris.shogun2.model.wps;
+
+import javax.persistence.Entity;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+
+/**
+ *
+ */
+@Entity
+@Table
+public class WpsReference extends WpsParameter {
+
+	/**
+	 *
+	 */
+	private static final long serialVersionUID = 1L;
+	
+	/**
+	 * 
+	 */
+	private String url;
+	
+	/**
+	 * 
+	 */
+	private String mimeType;
+	
+	/**
+	 * 
+	 */
+	private String method;
+
+	/**
+	 *
+	 */
+	@ManyToOne
+	private WpsParameter body;
+
+	/**
+	 * Constructor
+	 */
+	public WpsReference() {
+	}
+
+
+	/**
+	 * @return the url
+	 */
+	public String getUrl() {
+		return url;
+	}
+
+
+	/**
+	 * @param url the url to set
+	 */
+	public void setUrl(String url) {
+		this.url = url;
+	}
+
+
+	/**
+	 * @return the mimeType
+	 */
+	public String getMimeType() {
+		return mimeType;
+	}
+
+
+	/**
+	 * @param mimeType the mimeType to set
+	 */
+	public void setMimeType(String mimeType) {
+		this.mimeType = mimeType;
+	}
+
+
+	/**
+	 * @return the method
+	 */
+	public String getMethod() {
+		return method;
+	}
+
+
+	/**
+	 * @param method the method to set
+	 */
+	public void setMethod(String method) {
+		this.method = method;
+	}
+
+
+	/**
+	 * @return the body
+	 */
+	public WpsParameter getBody() {
+		return body;
+	}
+
+
+	/**
+	 * @param body the body to set
+	 */
+	public void setBody(WpsParameter body) {
+		this.body = body;
+	}
+
+
+	/**
+	 * @see java.lang.Object#hashCode()
+	 *
+	 * According to
+	 * http://stackoverflow.com/questions/27581/overriding-equals-and-hashcode-in-java
+	 * it is recommended only to use getter-methods when using ORM like Hibernate
+	 */
+	@Override
+	public int hashCode() {
+		return new HashCodeBuilder(11, 47) // two randomly chosen prime numbers
+			.appendSuper(super.hashCode())
+			.append(getUrl())
+			.append(getMimeType())
+			.append(getMethod())
```

Comment:
I think the information used here (`url`, `mimeType`, `method` and the properties from the parent class) should be enough to make this *unique*. Leaving the `body` out could increase performance as lazy loading is avoided...